### PR TITLE
feat(iotda): support device proxy resource

### PR DESCRIPTION
**Pull request by jinyangyang222**
**What this PR does / why we need it**:

Support device proxy resource

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:
**Release note**:

```release-note

Support device proxy resource

```

## PR Checklist

* [x] Tests added/passed.

```
$ export HW_IOTDA_ACCESS_ADDRESS=xxxxxxxxxxxxx

$ make testacc TEST="./huaweicloud/services/acceptance/iotda" TESTARGS="-run TestAccDeviceProxy_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iotda -v -run TestAccDeviceProxy_basic -timeout 360m -parallel 4
=== RUN   TestAccDeviceProxy_basic
--- PASS: TestAccDeviceProxy_basic (21.32s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iotda     21.384s

```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
image

  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
image

### CODE REVIEW

File: huaweicloud/services/iotda/resource_huaweicloud_iotda_device_proxy.go [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+					},
+				},
+			},
+		},
+	}
+}
+
+func buildEffectiveTimeRangeBodyParams(d *schema.ResourceData) *model.EffectiveTimeRange {
+	return &model.EffectiveTimeRange{
+		StartTime: utils.String(d.Get("effective_time_range").([]interface{})[0].(map[string]interface{})["start_time"].(string)),
```
Comment (deer-hang) [inline huaweicloud/services/iotda/resource_huaweicloud_iotda_device_proxy.go]: Risk of null pointer.

**Comment by jinyangyang222** [comment]
make testacc TEST="./huaweicloud/services/acceptance/iotda" TESTARGS="-run TestAccDeviceProxy_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iotda -v -run TestAccDeviceProxy_basic -timeout 360m -parallel 4
=== RUN   TestAccDeviceProxy_basic
--- PASS: TestAccDeviceProxy_basic (20.57s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iotda     20.609s

File: huaweicloud/services/iotda/resource_huaweicloud_iotda_device_proxy.go (hunk 1 above)
Comment (jinyangyang222) [reply huaweicloud/services/iotda/resource_huaweicloud_iotda_device_proxy.go]: fixed.

**Pull request merged**